packages/capabilities/src/consumer.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Provider Capabilities
+ *
+ * These can be imported directly with:
+ * ```js
+ * import * as Consumer from '@web3-storage/capabilities/consumer'
+ * ```
+ *
+ * @module
+ */
+import { capability, DID, URI, Link } from '@ucanto/validator'
+// @ts-ignore
+// eslint-disable-next-line no-unused-vars
+import * as Types from '@ucanto/interface'
+import { equalWith, fail, equal, equalCID } from './utils.js'
+import { top } from './top.js'
+
+export { top }
+
+/**
+ * Capability can only be delegated (but not invoked) allowing audience to
+ * derived any `consumer/` prefixed capability for the agent identified
+ * by did:key in the `with` field.
+ */
+export const consumer = top.derive({
+  to: capability({
+    can: 'consumer/*',
+    with: DID,
+    derives: equalWith,
+  }),
+  derives: equalWith,
+})
+
+const base = top.or(consumer)
+
+/**
+ * Capability can be invoked by an agent to request a `consumer/add` for an account.
+ */
+export const add = base.derive({
+  to: capability({
+    can: 'consumer/add',
+    /**
+     * Must be an provider DID
+     */
+    with: DID,
+    nb: {
+      /**
+       * CID for the provider/get invocation
+       */
+      request: Link,
+      /**
+       * Support specific space DIDs or undefined to request a provider for multiple spaces
+       */
+      consumer: URI.match({ protocol: 'did:' }).optional(),
+    },
+    derives: (child, parent) => {
+      return (
+        fail(equalWith(child, parent)) ||
+        fail(equalCID(child.nb.request, parent.nb.request, 'request')) ||
+        fail(equal(child.nb.consumer, parent.nb.consumer, 'consumer')) ||
```

Comment:
It seems like to [accomodate this from spec](https://github.com/web3-storage/specs/blob/576b988fb7cfa60049611963179277c420605842/w3-provider.md?plain=1#L211), this check might need to adjust to authorize deriving from `nb.consumer: did:*` (or consumer undefined, which spec [says](https://github.com/web3-storage/specs/blob/576b988fb7cfa60049611963179277c420605842/w3-provider.md?plain=1#L234) is equivalent) to `nb.consumer: did:key:xyz`